### PR TITLE
Update advanced TOC section to work with nested groups

### PR DESCRIPTION
**Pull request by spatialthoughts**
Updating the example to find groups and nodes to work with nested groups. The proposed change uses a recursive function to get all child groups.

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/cheat_sheet.rst [hunk 1]
```diff
@@ -459,14 +459,25 @@ Advanced TOC
...
-    for child in root.children():
-        if isinstance(child, QgsLayerTreeGroup):
-            print ("- group: " + child.name())
-        elif isinstance(child, QgsLayerTreeLayer):
-            print ("- layer: " + child.name())
-
+   from qgis.core import QgsLayerTreeGroup, QgsLayerTreeLayer
+   
+   def getGroupLayers(group):
+      print('- group:' + group.name())
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]:
```suggestion
      print('- group: ' + group.name())
```

Comment (havatv) [reply source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]: To make travis happy

File: source/docs/pyqgis_developer_cookbook/cheat_sheet.rst [hunk 2]
```diff
@@ -459,14 +459,25 @@ Advanced TOC
...
+      for child in group.children():
+         if isinstance(child, QgsLayerTreeGroup):
+            # Recursive call to get nested groups
+            getGroupLayers(child)
+         else:
+            print('  - layer:' + child.name())
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]:
```suggestion
            print('  - layer: ' + child.name())
```

Comment (spatialthoughts) [reply source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]: Thanks. Fixed.

File: source/docs/pyqgis_developer_cookbook/cheat_sheet.rst [hunk 3]
```diff
@@ -459,14 +459,25 @@ Advanced TOC
...
-
-    for child in root.children():
-        if isinstance(child, QgsLayerTreeGroup):
-            print ("- group: " + child.name())
-        elif isinstance(child, QgsLayerTreeLayer):
-            print ("- layer: " + child.name())
-
+   from qgis.core import QgsLayerTreeGroup, QgsLayerTreeLayer
+   
+   def getGroupLayers(group):
```
Comment (Gustry) [inline source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]:
```suggestion
   def get_group_layers(group):
```

File: source/docs/pyqgis_developer_cookbook/cheat_sheet.rst [hunk 4]
```diff
@@ -459,14 +459,25 @@ Advanced TOC
...
+      print('- group: ' + group.name())
+      for child in group.children():
+         if isinstance(child, QgsLayerTreeGroup):
+            # Recursive call to get nested groups
+            getGroupLayers(child)
```
Comment (Gustry) [inline source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]:
```suggestion
            get_group_layers(child)
```

File: source/docs/pyqgis_developer_cookbook/cheat_sheet.rst [hunk 5]
```diff
@@ -459,14 +459,25 @@ Advanced TOC
...
+         else:
+            print('  - layer: ' + child.name())
+
+
+   root = QgsProject.instance().layerTreeRoot()
+   for child in root.children():
+      if isinstance(child, QgsLayerTreeGroup):
+         getGroupLayers(child)
```
Comment (Gustry) [inline source/docs/pyqgis_developer_cookbook/cheat_sheet.rst]:
```suggestion
         get_group_layers(child)
```

**Comment by spatialthoughts** [comment]
This is ready to be merged. Let me know if any action is required from my side.

**Comment by havatv** [comment]
Are there any reasons for not backporting this to 3.4?

**Comment by spatialthoughts** [comment]
No. This can be applied to 3.4. Do I need to change anything in the PR?

**Comment by havatv** [comment]
Thanks, @spatialthoughts! I have added the backport label and merged.

**Comment by Gustry** [comment]
> Do I need to change anything in the PR?

It was in the PR template that you removed ;-)

**Pull request merged**